### PR TITLE
Skip slots v2

### PR DESCRIPTION
**Pull request by tukan**

### CODE REVIEW

**Comment by hawkw** [comment]
hi @tukan, what's the relationship between this PR and your previous #80 ? are you still planning to work on #80 as well?

File: src/lib.rs
Comment (hawkw) [inline src/lib.rs:556]: can't this just be a `fetch_and` operation to clear the bit, rather than `fetch_update` (which performs a CAS under the hood):
```suggestion
            test_dbg!(self
                .slot
                .state
                .fetch_and(!(1 << (usize::BITS - 1)), SeqCst))
```
(this might look a bit nicer if we had a constant for `1 << (usize::BITS - 1))`...)

File: src/lib.rs
Comment (hawkw) [reply src/lib.rs:110]: It might be worth having a comment here that describes the use of the other bits in this field, as well, but since I didn't already write that down, I won't block this PR on making you do it for me. :)

**Review by hawkw — COMMENTED** [review]
Overall, I definitely prefer this approach over the one proposed in #80. I had a handful of minor suggestions, most of which are just stylistic --- feel free to ignore most of them.

I do think we should ensure the new slow loom models are added to the list of slow models that run in separate CI jobs, so that we can run those tests on CI in the future.

File: src/lib.rs
Comment (tukan) [reply src/lib.rs]: Yes, so I think we can use your readable ```state & HAS_READER == HAS_READER```:  https://godbolt.org/z/sadzs7jhr

File: src/lib.rs
Comment (tukan) [reply src/lib.rs]: My reasoning was not to have an inner loop :) but in this case, spurious failure is expensive indeed 👍

File: src/lib.rs
Comment (tukan) [reply src/lib.rs:408]: agree, no sense in backing off here 👍

File: src/lib.rs
Comment (tukan) [reply src/lib.rs]: Looks nice, I just changed formatting to be compatible with 1.57.0

File: src/lib.rs
Comment (tukan) [reply src/lib.rs]: Done  :) https://godbolt.org/z/dqeoM3ndd

File: src/lib.rs
Comment (tukan) [reply src/lib.rs:50]: Will do 👍

File: src/lib.rs
Comment (hawkw) [reply src/lib.rs]: oh whoa, debug more is substantially more code, but release is just one extra instruction!

**Comment by tukan** [comment]
Hello @hawkw,

I hope you're well. I wanted to kindly bring to your attention this PR. Could you please finish reviewing it when you have a moment. Thank you.

**Comment by hawkw** [comment]
Hi, I'm sorry, I must have missed this earlier! I'd love to get this PR merged soon and I'll give it another review shortly.

File: src/lib.rs
Comment (hawkw) [inline src/lib.rs:484]: There are now enough operations on `state` that I wonder if it's worth turning it into a newtype...but, we can do that in a separate PR.

**Review by hawkw — APPROVED** [review]
Okay, this looks good to me! I'm going to go ahead and merge it now. Thanks @tukan for the fix!

**Pull request merged**
